app/slave/cluster_slave.py

```diff
@@ -310,9 +310,13 @@ def _execute_subjob(self, build_id, subjob_id, executor, atomic_commands):
         files = {'file': ('payload', open(results_file, 'rb'), 'application/x-compressed')}
 
         self._idle_executors.put(executor)  # work is done; mark executor as idle
-        self._network.post(results_url, data=data, files=files)  # todo: check return code
-
-        self._logger.info('Build {}, Subjob {} completed and sent results to master.', build_id, subjob_id)
+        resp = self._network.post(results_url, data=data, files=files)
+        if resp.ok:
+            self._logger.info('Build {}, Subjob {} completed and sent results to master.', build_id, subjob_id)
+        else:
+            self._logger.error(
+                ('Build {}, Subjob {} encountered an error when sending results to master.'
+                 '\n\tStatus Code {}\n\t{}').format(build_id, subjob_id, resp.status_code, resp.text))
```

Comment:
Not sure if any of our other logs contain new lines (except for stack traces) but this is probably fine.